File: Python/Codes/Practice/LeetCode/495 Teemo Attacking.py

class Solution(object):
    def findPoisonedDuration(self, timeSeries, duration):
        """
        :type timeSeries: List[int]
        :type duration: int
        :rtype: int
        """

        sumDuration = 0
        lastEnd = 0        
        for v in timeSeries:
            dValue = duration
            if v < lastEnd:
                dValue += v - lastEnd
            
            sumDuration += dValue
            lastEnd = v + duration

        return sumDuration


if __name__ == '__main__':
    s = Solution()
    print(s.findPoisonedDuration([1,4],2))
    print(s.findPoisonedDuration([1,2],2))
    print(s.findPoisonedDuration([1,4,5,6,10,14,15,34],2))
    

File: CSharp/Codes/Practice/LeetCode/495 Teemo Attacking.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpCodes.Codes.Practice.LeetCode
{
    class Solution
    {
        public int FindPoisonedDuration(int[] timeSeries, int duration)
        {
            int sumDuration = 0;
            int lastEnd = 0;
            foreach(int v in timeSeries)
            {
                int dValue = duration;
                if (v < lastEnd)
                {
                    dValue += v - lastEnd;
                }

                sumDuration += dValue;
                lastEnd = v + duration;
            }

            return sumDuration;
        }
    }

    [TestClass()]
    public class TeemoAttackingTests
    {
        [TestMethod()]
        public void FindPoisonedDuration()
        {
            var s = new Solution();
            Assert.AreEqual(s.FindPoisonedDuration(new int[] { 1, 4 }, 2), 4);
        }
    }
}